terraform/services/ecr-synk-integration/main.tf

```diff
@@ -0,0 +1,99 @@
+
+locals {
+  repos = {
+    ab2d = [
+      "repo:CMSgov/ab2d-bcda-dpc-platform:*",
+      "repo:CMSgov/ab2d-events:*",
+      "repo:CMSgov/ab2d-lambdas:*",
+      "repo:CMSgov/ab2d:*",
+    ]
+    bcda = [
+      "repo:CMSgov/ab2d-bcda-dpc-platform:*",
+      "repo:CMSgov/bcda-app:*",
+    ]
+    dpc = [
+      "repo:CMSgov/ab2d-bcda-dpc-platform:*",
+      "repo:CMSgov/dpc-app:*",
+    ]
+  }
+  admin_app = var.app == "dpc" ? "bcda" : var.app
+}
+
+resource "aws_ssm_parameter" "snyk_integration_role_arn" {
+  name  = "/${var.app}-snyk/role-arn"
+  type  = "String"
+  value = aws_iam_role.snyk.arn
+}
+
+
+data "aws_ssm_parameter" "snyk_integration_role_arn" {
+  name = "/snyk-integration/role-arn"
+}
+
+data "aws_iam_role" "admin" {
+  name = "ct-ado-${local.admin_app}-application-admin"
+}
+
+data "aws_iam_policy_document" "snyk_role_policy" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+      "sts:TagSession",
+    ]
+
+    principals {
+      type = "AWS"
+      identifiers = [
+        data.aws_ssm_parameter.snyk_integration_role_arn.value,
+        data.aws_iam_role.admin.arn,
+      ]
+    }
+  }
+
+  statement {
+    sid       = "SnykAllowPull"
+    effect    = "Allow"
+    actions   = [
+      "ecr:GetLifecyclePolicyPreview",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:BatchGetImage",
+      "ecr:DescribeImages",
+      "ecr:GetAuthorizationToken",
+      "ecr:DescribeRepositories",
+      "ecr:ListTagsForResource",
+      "ecr:ListImages",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetRepositoryPolicy",
+      "ecr:GetLifecyclePolicy"
+    ]
+    resources = ["*"]
+  }
```

Comment:
This SnykAllowPull statement has to go in a separate policy document that's associated with the role via a new aws_iam_role_policy_attachment resource. Note the difference between an IAM role policy and the trust relationship, which is what the assume_role_policy sets below.